sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidget.kt

```diff
@@ -8,24 +8,20 @@ import com.mapbox.maps.MapboxExperimental
  *
  * @param bitmap bitmap used to draw widget
  * @param position position of widget
- * @param marginX horizontal margin in pixels
- * @param marginY vertical margin in pixels
  */
 @MapboxExperimental
 open class BitmapWidget @JvmOverloads constructor(
   bitmap: Bitmap,
   position: WidgetPosition = WidgetPosition(
     vertical = WidgetPosition.Vertical.TOP,
     horizontal = WidgetPosition.Horizontal.LEFT,
-  ),
-  marginX: Float = 0f,
-  marginY: Float = 0f,
+    offsetX = 0f,
+    offsetY = 0f
+  )
```

Comment:
I like the idea to move margin into the WidgetPosition. but end of the day i can see that we will continue to have backwards compatible issues.

For example, what will happen when we want to set the initial rotation? We will need to add a new breaking change in order to do so.

For this reason, I think we can make this constructor internal and make something else build the widget. CameraOptions is a good example of being backwards compatible and allowing future changes.

The equivalent for this class would be `BitmapWidgetOptions`, put everything into it.

Reply:
I'm not sure I agree with having a `BitmapWidgetOptions` where we add all the options for the `BitmapWidget`. I somehow would prefer more granular that could be re-used in other widgets.
For example, we could add a `Builder` for the `WidgetPosition`.
Wdyt?

Reply:
agree, we can use Builder for WidgetPosition 👍

Reply:
I'd also prefer not to merge all the configurations to the `BitmapWidgetOptions`, as some of the configuration like position will be more universal, and could be reused in other places. E.g. the main benefits of the current proposal is to be able to update the `WidgetPosition` after the initialisation of the widget.